docs/modules/ROOT/pages/how-to-guides/Secure-your-supply-chain/proc_java_dependencies.adoc

```diff
@@ -1,30 +1,110 @@
 
-* If all the builds are successful, your application's SBOM displays that none of the dependencies came from third-party repositories. However, you may need to troubleshoot when some dependencies fail to build.
+- The system analyzes the repository and uses the results to find a build strategy. For example, if the system cannot find the JDK version to use, the system uses all versions to determine the most effective version to use.
 
-.*Procedure*
+- The system attempts to build all the dependencies and stores them in container images in an image registry, and `quay.io` is the default.
+
+- When the system has completed all its builds, you can rebuild your application.
+
+- If all the builds are successful, your application's SBOM displays that none of the dependencies came from third-party repositories. However, you may need to troubleshoot when some dependencies fail to build.
+
+== Setup
+
+=== Logging into the Server
+
+All interactions with the JVM Build Service require you to be logged into the OpenShift namespace used for your builds.
+
+.Procedure
+
+Copy the login command from: https://registration-service-toolchain-host-operator.apps.stone-prd-host1.wdlc.p1.openshiftapps.com/
+
+=== Setting up the CLI
+
+Many of the commands below can be setup using the JVM Build Service CLI.
+
+The CLI is currently distributed as a docker image. To use the docker image, create the following alias:
+
+```
+alias jbs='docker run --mount type=bind,source=$HOME/.kube/config,target=/kube --mount type=bind,source=$HOME/.github,target=/root/.github --env KUBECONFIG=/kube  -it --rm   quay.io/redhat-appstudio/hacbs-jvm-cli:latest'
+```
+
+To update the latest version, create an update alias. You can run this alias to pull the latest version of the docker image:
+
+```
+alias update-jbs='docker  pull quay.io/redhat-appstudio/hacbs-jvm-cli:latest'
+```
+
+The CLI supports tab completion and all the commands have a --help option to show usage.
+
+Note that this CLI also requires you to be logged in as described above.
+
+To use commands that modify build recipes (i.e. commands that are used to fix failing builds) you will need to create a `$HOME/.github` file as specified at  https://github-api.kohsuke.org/[].
+
+== Configuring The JVM Rebuild Service
+
+There are three ways to setup the JVM Build Service, they are listed below.
+
+=== Setting up using the CLI
+
+If you do not require explicit quay.io configuration, and you have setup the CLI this is the simplest way to setup the JVM Build Service. To do this run the `setup rebuilds` command in the CLI:
+
+[source]
+----
+jbs> setup rebuilds
+Working...
+found .dockerconfigjson secret with appropriate token keys in namespace sdouglas1-tenant, rebuilds are possible
+Rebuilds setup successfully
+----
+
+This will automatically perform the steps detailed below and setup rebuilds for you. The JVM Build Service will automatically create a quary repository to store your rebuilt artifacts.
```

Comment:
this what? command?

use present instead of future--> This ______ automatically performs the steps...and sets up rebuilds for you. The... also automatically creates a....

is "quary" the right word? _query_?

also, we should avoid using directional words like "below" for two reasons:

1. The location could change depending on future modularization or reorg efforts.
2. I _think_ it's a no-no for accessibility.